release/models/nat/openconfig-nat.yang

```diff
@@ -0,0 +1,965 @@
+
+      list pool {
+        key "name";
+        description
+          "List of NAT address pools";
+
+        leaf name {
+          type leafref {
+            path "../config/name";
+          }
+          description
+            "Reference to pool name";
+        }
+
+        container config {
+          description
+            "Configuration parameters for the NAT pool";
+
+          uses nat-pool-config;
+        }
+
+        container state {
+          config false;
+          description
+            "Operational state data for the NAT pool";
+
+          uses nat-pool-config;
+          uses nat-pool-state;
+        }
+      }
+    }
+  }
+
+  grouping nat-mapping-top {
+    description
+      "Top-level grouping for static NAT mappings";
+
+    container mappings {
+      description
+        "Static NAT mapping configuration and state";
+
+      list mapping {
+        key "name";
+        description
+          "List of static NAT mappings";
+
+        leaf name {
+          type leafref {
+            path "../config/name";
+          }
+          description
+            "Reference to mapping name";
+        }
+
+        container config {
+          description
+            "Configuration parameters for the static mapping";
+
+          uses nat-mapping-config;
+        }
+
+        container state {
+          config false;
+          description
+            "Operational state data for the static mapping";
+
+          uses nat-mapping-config;
+          uses nat-mapping-state;
+        }
+      }
+    }
+  }
+
+  grouping nat-top {
+    description
+      "Top-level grouping for NAT configuration and state";
+
+    container nat {
+      description
+        "Top-level container for NAT";
+
+      container config {
+        description
+          "Global NAT configuration parameters";
+
+        leaf enabled {
+          type boolean;
+          description
+            "Global administrative state for NAT functionality";
+        }
+
+        leaf translation-timeout {
+          type uint32;
+          units "seconds";
+          description
+            "Default translation timeout for all NAT instances";
+        }
+
+        leaf max-translations {
+          type uint32;
+          description
+            "Maximum number of concurrent translations globally";
+        }
+
+        leaf log-level {
+          type log-level;
+          description
+            "Global NAT logging level";
+        }
+      }
+
+      container state {
+        config false;
+        description
+          "Global NAT operational state data";
+
+        leaf enabled {
+          type boolean;
+          description
+            "Global administrative state for NAT functionality";
+        }
+
+        leaf translation-timeout {
+          type uint32;
+          units "seconds";
+          description
+            "Default translation timeout for all NAT instances";
+        }
+
+        leaf max-translations {
+          type uint32;
```

Comment:
what happens when a new traffic flow is presented to the NAT subsystem that would normally be translated but we are already at this limit?

Reply:
I think that's up to the vendor implementation. Arista, for example, states:
```
The ip nat translation max-entries command specifies maximum number of NAT translation connections. After this threshold is reached, new connections are dropped until the number of programmed connections is reduced below the level specified by the ip nat translation low-mark command.
```

SOURCE: https://www.arista.com/en/um-eos/eos-data-plane-security#xx1149232

Reply:
Can we introduced a configuration and enum that enumerates this scenario? The behavior should always be explicitly defined.